okio/src/main/java/okio/Buffer.java

```diff
@@ -1263,18 +1267,23 @@ Segment writableSegment(int minimumCapacity) {
   }
 
   @Override public long indexOf(byte b) {
-    return indexOf(b, 0);
+    return indexOf(b, 0, Long.MAX_VALUE);
   }
 
   /**
    * Returns the index of {@code b} in this at or beyond {@code fromIndex}, or
    * -1 if this buffer does not contain {@code b} in that range.
    */
   @Override public long indexOf(byte b, long fromIndex) {
+    return indexOf(b, fromIndex, Long.MAX_VALUE);
+  }
+
+  @Override public long indexOf(byte b, long fromIndex, long toIndex) {
     if (fromIndex < 0) throw new IllegalArgumentException("fromIndex < 0");
 
     Segment s;
     long offset;
+    if (toIndex > size) toIndex = size;
```

Comment:
After some more thought, I don't think throwing is appropriate. The buffer may not be populated with all of the stream's future data yet, even though the caller still knows that they're willing to read up to n bytes. This happens in RealBufferedSource.indexOf, where additional data may be read from the source if the target byte isn't found yet.